### DISCUSSION TITLE
Regex revision for linuxserver.io containers - no update detected when the capture groups major,minor,patch,revision are used instead of the capture groups major,minor,patch,build

### DISCUSSION THREAD

**Discussion by kennethso168** [renovatebot/renovate#33270] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

For the following image:

`image: ghcr.io/linuxserver/librespeed:5.4.1-ls215`

and it is known that the latest version should be 5.4.1-ls218

No PR is opened with the regex `"versioning": "regex:^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)-ls(?<revision>.+)$"`.

However, such PR can be successfully opened with the regex `"versioning": "regex:^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)-ls(?<build>.+)$"`.

As the regex is [adapted from an example from the docs](https://docs.renovatebot.com/modules/versioning/regex/#description), at least the docs should be updated if this is not the intended behavior.

[Minimal reproduction repository](https://github.com/kennethso168/linuxserver-io-librespeed-repr-20241224)

Related discussion: https://github.com/renovatebot/renovate/discussions/33024

### Logs (if relevant)

The logs has been uploaded to the [Minimal reproduction repository](https://github.com/kennethso168/linuxserver-io-librespeed-repr-20241224) as well

**Comment by viceice** [comment, marked as answer]
the docs clearly say `revision` can be used after `build` was used. so you can't use it without.

**Comment by kennethso168** [comment]
Ok, that means the example in the docs are wrong, as mentioned in #33024 . Lemme see if I could do a PR for the docs when I have some free time. Thanks and happy holidays!

**Comment by kennethso168** [comment]
Sorry @viceice upon further reading the docs it seems that I'm wrong to say that the example in the docs are wrong. The example uses `ghcr.io/linuxserver/openssh-server` which really contains five version parts so that all `major`, `minor`, `patch`, `build` and `revision` capture groups are needed. #33024 should be another issue. I'll try to do another minimal repo for that to see if that issue is a bug or related to that user's self hosted configuration.

I'd like to update that doc to fix the typo anyway. I think most linuxserver.io image tags follow semver plus build number (`x.y.z-lsnn`). `ghcr.io/linuxserver/openssh-server` is a particularly difficult case with the five capture groups. Do you think it is a good idea to add an additional example to show how to handle linuxserver.io image tags in the format of `x.y.z-lsnn`? I believe that this will let beginners to get started to use renovate with most linuxserver.io images. This will also clearly demonstrate that when only four version parts are needed, `build` instead of `revision` should be used for the fourth capture group.

**Reply by viceice** [reply]
sure, PR welcome

**Reply by kennethso168** [reply]
Already done in #33333
